File: historian-core/src/main/java/net/yslibrary/historian/internal/LogTable.java
package net.yslibrary.historian.internal;

/**
 * Table definition of Log
 */

class LogTable {

  public static final String NAME = "log";

  @SuppressWarnings("StringBufferReplaceableByString")
  public static final String CREATE_TABLE = new StringBuilder()
      .append("CREATE TABLE ").append(NAME)
      .append(" (")
      .append("id INTEGER PRIMARY KEY AUTOINCREMENT,")
      .append("priority TEXT NOT NULL, ")
      .append("tag TEXT NOT NULL, ")
      .append("message TEXT NOT NULL, ")
      .append("created_at INTEGER NOT NULL")
      .append(");")
      .toString();

  @SuppressWarnings("StringBufferReplaceableByString")
  public static final String INSERT = new StringBuilder()
      .append("INSERT INTO ").append(NAME)
      .append("(priority, tag, message, created_at) ")
      .append("VALUES(?, ?, ?, ?);")
      .toString();

  @SuppressWarnings("StringBufferReplaceableByString")
  public static final String DELETE_OLDER = new StringBuilder()
      .append("DELETE FROM ").append(NAME)
      .append(" where id NOT IN (")
      .append("SELECT id FROM log ORDER BY created_at DESC LIMIT ?")
      .append(");")
      .toString();

  private LogTable() {
    // no-op
  }
}

File: historian-core/src/main/java/net/yslibrary/historian/internal/LogTable.kt
package net.yslibrary.historian.internal

/**
 * Table definition of Log
 */
object LogTable {
    const val NAME = "log"

    val CREATE_TABLE = """
        CREATE TABLE $NAME (
            id INTEGER PRIMARY KEY AUTOINCREMENT,
            priority TEXT NOT NULL,
            tag TEXT NOT NULL,
            message TEXT NOT NULL,
            created_at INTEGER NOT NULL
        );
    """.trimIndent()

    val DROP_TABLE = "DROP TABLE $NAME;"

    val INSERT = """
        INSERT INTO $NAME(priority, tag, message, created_at) VALUES(?, ?, ?, ?);
    """.trimIndent()

    val DELETE_OLDER = """
        DELETE FROM $NAME WHERE id NOT IN (SELECT id FROM $NAME ORDER BY created_at DESC LIMIT ?);
    """.trimIndent()
}
